model_interfaces/interface.py

```diff
@@ -3,13 +3,17 @@
 from typing import Tuple, Callable
 from dataclasses import dataclass, field
 
+from utils.constants import PERSISTENCE_DIR, ResetPolicy
+
 
 @dataclass
 class ChatSession(ABC):
     history: list[dict[str, str | datetime]] = field(default_factory=list)
     costs_usd: float = 0
     is_local: bool = False
     max_message_size: int = 1000
+    dataset_name: str = ""
+    reset_policy: ResetPolicy = ResetPolicy.HARD
```

Comment:
What about making the reset policy soft by default, since that's the one that will be used mostly? To implement an agent one has to implement `save` and `load` anyway, so setting the reset policy could be part of it.